### DISCUSSION TITLE
Type Unknown must have [Symbol.iterator]() Context API

### DISCUSSION THREAD

**Discussion by anutting** [solidjs/solid#555] [Q&A]
New to Solidjs and relatively new to TypeScript. I'm following the tutorial and encountered this error when trying to use the Context API. The same error is showing up in my local IDE.

![Screen Shot 2021-07-09 at 9 42 26 AM](https://user-images.githubusercontent.com/21373974/125086987-1cefe000-e09a-11eb-9631-e12d6c95ee1c.png)

**Comment by anutting** [comment]
It looks like the Value prop of the <Context.Provider> component has a type of unknown.

**Comment by excessivecoding** [comment]
Not the best at TS or Solid for the matter.

First, it's important that Typescript "error" are really just warning, your counter should still work even if the REPL return Typescript error as there are stripped down when compiling (unless you activate the strict mode).

Here how I manage to deal with the warning/error.

https://playground.solidjs.com/?hash=1737220061&version=1.0.0

We need to tell Typescript what our Context value will look like, so it can understand it.
We can do that by specifying a type in `useContext` (cf. the like to the playground in the file `counter.tsx`).

Once Typescript understand what `useCounter` is returning, you should be good to go.

If you have question in the future, try to open a discussion in https://github.com/solidjs/solid/discussions as I think issues are meant for codebase related matter

**Reply by corbolt** [reply]
This bug seems still open/unresolved. both, my local ide as well as the playground editor show this issue and im not able to build.
